example-plugins/inj-tyfam-plugin/app/Main.hs

```diff
@@ -0,0 +1,43 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+
+{-# OPTIONS_GHC -fplugin=Plugin.InjTyFam #-}
+{-# OPTIONS_GHC -dcore-lint #-}
+
+module Main (main) where
+
+import Prelim
+
+main :: IO ()
+main = pure ()
+
+_id :: (Id a ~ Id b) => a -> b
+_id x = x
+
+_both :: (Both a x ~ Both b y) => (a, x) -> (b, y)
+_both x = x
+
+_rite :: (Rite a x ~ Rite b y) => (a, b, x) -> (a, b, y)
+_rite x = x
+
+-- | This fails to type-check if the plugin only emits the @ai ~ bi@
+-- constraints without the updated @F a1 .. an ~ F c1 .. cn@.
+_rite2 :: forall a b x y. (Rite a x ~ Rite b y) => (a, b, x) -> (a, b, y)
+_rite2 x = (() :: (Rite a y ~ Rite b x) => ()) `seq` x
+
+{-
+
+-- TODO figure out a robust yet lightweight mechanism to test for
+-- expected type errors
```

Comment:
I use [doctest](https://hackage.haskell.org/package/doctest):

    -- |
    -- >>> :{
    -- unit :: ()
    -- unit = requireMyNullary
    -- :}
    -- ...
    -- ... No instance for MyNullary
    -- ...

Reply:
Cool, I'll keep this in mind next time I need this functionality.

Reply:
Thanks for the tip. Let's keep this in mind if we ever dive into CI.